platforms/jvm/toolchains-jvm/src/main/java/org/gradle/jvm/toolchain/internal/install/JdkCacheDirectory.java

```diff
@@ -223,49 +247,34 @@ private static String getInstallFolderName(JvmInstallationMetadata metadata) {
         String os = OperatingSystem.current().getFamilyName();
         return String.format("%s-%s-%s-%s", vendor, version, architecture, os)
                 .replaceAll("[^a-zA-Z0-9\\-]", "_")
-                .toLowerCase();
+                .toLowerCase() + ".2";
```

Comment:
❓ Do we need the `.2` now that we're checking the install directory to see if it has a previous installation? With this `.2` I don't see how we would ever see a prior installed toolchain.

EDIT: This also seems to clash with the code below that writes the legacy marker file, as the old code would never see this

Reply:
So, this is kind of complicated. Because the old toolchain marker is flaky, we can't actually use a previous installation here, as it might not be valid. We also don't want to conflict with the old installation logic if it's running concurrently. I talked with Sterling and we decided that we should use a new directory name that can't possibly conflict; but we add a legacy marker as the installation detection logic doesn't use specifically formatted names, it will take any subfolder with an appropriate marker.

The end result is that 8.7 and 8.8 will use different installation paths, 8.8 won't use previously-installed 8.7 JDKs, but 8.7 will use newly installed 8.8 JDKs. However, I'm not sure the small chance of 8.7 and 8.8 running concurrently, installing concurrently, is enough to warrant a new directory like this.

Reply:
To me that means we don't need to check in this directory whether there is a partially installed distribution. Unless, we are looking for partially installed distributions that were somehow installed by this version of Gradle

Reply:
When there is no marker file there can still be cases (when atomic move is not supported) where the install folder copy starts, then the JDK is forcibly killed without being able to do clean up. So yes, there can still be partially installed distributions -- the difference is they will never have the marker file.